refactoringminer/src/main/java/gr/uom/java/xmi/decomposition/Visitor.java

```diff
@@ -36,707 +43,223 @@
-            }
-        }
-        return super.visit(node);
-    }
-
-    private boolean complexInvocation(OperationInvocation invocation) {
-        return (invocation.numberOfSubExpressions() > 3 && invocation.containsVeryLongSubExpression()) ||
-            invocation.numberOfSubExpressions() > 15;
-    }
-
-    public static String processMethodInvocation(MethodInvocation node) {
+    public static String processMethodInvocation(PsiMethodCallExpression node) {
         StringBuilder sb = new StringBuilder();
-        sb.append(node.getName().getIdentifier());
+        sb.append(node.getMethodExpression().getReferenceName());
         sb.append("(");
-        List<Expression> arguments = node.arguments();
-        if (arguments.size() > 0) {
-            for (int i = 0; i < arguments.size() - 1; i++)
-                sb.append(arguments.get(i).toString()).append(", ");
-            sb.append(arguments.get(arguments.size() - 1).toString());
+        PsiExpression[] arguments = node.getArgumentList().getExpressions();
+        if (arguments.length > 0) {
+            for (int i = 0; i < arguments.length - 1; i++)
+                sb.append(arguments[i].toString()).append(", ");
+            sb.append(arguments[arguments.length - 1].toString());
         }
         sb.append(")");
         return sb.toString();
     }
 
-    public boolean visit(SuperMethodInvocation node) {
-        List<Expression> arguments = node.arguments();
-        for (Expression argument : arguments) {
-            processArgument(argument);
-        }
-        OperationInvocation invocation = new OperationInvocation(cu, filePath, node);
-        String nodeAsString = node.toString();
-        if (methodInvocationMap.containsKey(nodeAsString)) {
-            methodInvocationMap.get(nodeAsString).add(invocation);
-        } else {
-            List<OperationInvocation> list = new ArrayList<>();
-            list.add(invocation);
-            methodInvocationMap.put(nodeAsString, list);
-        }
-        if (current.getUserObject() != null) {
-            AnonymousClassDeclarationObject anonymous = (AnonymousClassDeclarationObject) current.getUserObject();
-            Map<String, List<OperationInvocation>> anonymousMethodInvocationMap = anonymous.getMethodInvocationMap();
-            if (anonymousMethodInvocationMap.containsKey(nodeAsString)) {
-                anonymousMethodInvocationMap.get(nodeAsString).add(invocation);
-            } else {
-                List<OperationInvocation> list = new ArrayList<>();
-                list.add(invocation);
-                anonymousMethodInvocationMap.put(nodeAsString, list);
+    @Override
+    public void visitElement(@NotNull PsiElement element) {
+        boolean goInSubtree = true;
+        if (element instanceof PsiArrayAccessExpression) {
+            String source = Formatter.format(element);
+            arrayAccesses.add(source);
+            if (!stackAnonymous.isEmpty()) {
```

Comment:
why do we check stackAnonymous here and in the checkings below?

Reply:
Logic or style? 
Logic is that we want to remove statements of Anonymous Classes in some places and we want to check if we in it. 
Style is that extracting it to the variable not makes code shorter or simpler. And some cases proceeds in extracted function so they need to check that condition again